### PR TITLE
As per #356

### PR DESCRIPTION
**Pull request by RlonRyan**
More Javadoc and minor code improvements.

### CODE REVIEW

File: src/main/java/com/InfinityRaider/AgriCraft/apiimpl/v1/cropplant/CropPlant.java [hunk 1]
```diff
@@ -36,63 +49,175 @@ public final int getTier() {
...
+     * @param rand a random for choosing the drop. Should eventually be replaced by {@link Constants#RAND}.
+     * @return a random fruit dropped by the plant.
+     */
     public abstract ItemStack getRandomFruit(Random rand);
 
-    /** Returns an ArrayList with amount of random fruit stacks for this plant */
+    /**
+     * Returns an ArrayList with amount of random fruit stacks for this plant.
+     * 
+     * @param gain the gain, as in number of drops. Should eventually be removed as the plant should already know this.
```
Comment (InfinityRaider) [inline src/main/java/com/InfinityRaider/AgriCraft/apiimpl/v1/cropplant/CropPlant.java]: Gain is passed to allow for more "rare" fruits for higher gain levels, as  seen with the ChocoCraft Gyshals

File: src/main/java/com/InfinityRaider/AgriCraft/utility/IOHelper.java [hunk 2]
```diff
@@ -321,7 +321,7 @@ public static ItemStack getStack(String input) {
             "harvestcraft:kiwiseedItem=harvestcraft:cantaloupeseedItem+harvestcraft:strawberryseedItem\n" +
             "harvestcraft:leekseedItem=harvestcraft:scallionseedItem+harvestcraft:celeryseedItem\n" +
             "harvestcraft:lettuceseedItem=AgriCraft:seedDaisy+harvestcraft:celeryseedItem\n" +
-            "harvestcraft:spinachseedItem=harvestcraft:lettuceseedItem+AgriCraft:seedCactus\n" +
+            "harvestcraft:spinachseedItem=harvestcraft:lettuceseedItem+AgriCraft:seedCactus\n" + // This line seems not to be valid, in my testing.
```
Comment (InfinityRaider) [inline src/main/java/com/InfinityRaider/AgriCraft/utility/IOHelper.java]: Spinach got added in a later version of Harvestcraft, and I was too lazy to update it in my dev env.

**Comment by InfinityRaider** [comment]
Thanks, appreciate the effort.

File: src/main/java/com/InfinityRaider/AgriCraft/farming/CropProduce.java [hunk 3]
```diff
@@ -84,6 +84,7 @@ public int getWeight(ItemStack stack) {
     }
 
 
+    // Something funny is going on here.
```
Comment (InfinityRaider) [inline src/main/java/com/InfinityRaider/AgriCraft/farming/CropProduce.java]: Like what?

Comment (RlonRyan) [inline src/main/java/com/InfinityRaider/AgriCraft/farming/CropProduce.java]: Some of the field appear as if they are never used, and the inner class seems under-utilized to be there.

Comment (InfinityRaider) [inline src/main/java/com/InfinityRaider/AgriCraft/farming/CropProduce.java]: minGain isn't used, it's there for if I ever decide to implement crops which require a certain minimum gain before they can start dropping certain fruits.
The inner class is just to be able to quickly access and change it without having to work with mulitple Lists or Maps.

Comment (RlonRyan) [inline src/main/java/com/InfinityRaider/AgriCraft/farming/CropProduce.java]: Ah. Ok.

**Pull request merged**